core/src/main/java/org/jsmart/zerocode/core/engine/mocker/RestEndPointMocker.java

```diff
@@ -109,10 +109,13 @@ private static MappingBuilder createRequestBuilderWithHeaders(MockStep mockStep,
     }
 
     private static ResponseDefinitionBuilder responseBuilder(MockStep mockStep, String jsonBodyRequest) {
-        final ResponseDefinitionBuilder responseBuilder = aResponse()
-                .withStatus(mockStep.getResponse().get("status").asInt())
-                //.withHeader("Content-Type", APPLICATION_JSON) //Can be XMl also for SOAP end points
-                .withBody(jsonBodyRequest);
+        ResponseDefinitionBuilder responseBuilder = aResponse()
+                .withStatus(mockStep.getResponse().get("status").asInt());
+        JsonNode headers = mockStep.getResponse().get("headers");
+        JsonNode contentType = headers != null?headers.get("Content-Type"):null;
+        responseBuilder = contentType != null?
+                responseBuilder.withHeader("Content-Type", contentType.toString()).withBody(jsonBodyRequest):
```

Comment:
Changes look good.
Can you please cover this via a unit test?
THere will be existing tests which can reuse and create a new one.

Reply:
Added Unit Test